### ISSUE TITLE
Add missing Selenium browsers in Citrus integration

### ISSUE THREAD

**Issue by svettwer** [citrusframework/citrus#498]
**User story**
As a citrus test developer, I want to be able to use all Browsers provided by selenium in my citrus test cases, so that I'm able to use all browsers to test my application.

**Additional context**
Derived from #495.
Additional refinement required.

**Comment by christophd** [comment]
I would suggest to use `custom` browser type. Isn't that a solution?

    <citrus-selenium:browser id="mySeleniumBrowser"
                type="custom"
                web-driver="operaWebDriver"/>

    <bean id="operaWebDriver" class="org.openqa.selenium.opera.OperaDriver"/>

**Comment by SajanaKrishnan** [comment]
Can you help me with java dsl?

**Comment by christophd** [comment]
Of course

    @Bean
    public SeleniumBrowser browser() {
        return CitrusEndpoints.selenium()
                .browser()
                .type("custom")
                .webDriver(driver())
                .build();
    }

    @Bean
    public WebDriver driver() {
        return new OperaDriver();
    }

**Comment by SajanaKrishnan** [comment]
Thanks. Let me check this and get back. In this how do I add desired
capabilities?

**Comment by christophd** [comment]
You may need to find proper Selenium browser implementation provided by Selenium itself and add it to the dependencies section in your Maven POM (pom.xml). Configuration of the browser implementation depends on the browser implementation API. So you may have a look at the browser implementation documentation provided by Selenium.

**Comment by SajanaKrishnan** [comment]
Thanks Chris
 Let me get back after I am done trying this.
Can you pls help resolve my other problem #493?

**Comment by svettwer** [comment]
Hi @christophd!

Yeah, the `custom` browser was my advice as well. Nevertheless it would be nice to match the out of the box browsers with the `BrowserType` enumeration provided by selenium. This would provide some intuitive behavior to users familiar with selenium.

**Comment by svettwer** [comment]
Fair enough. That's why I mentioned that more refinement is required before this issue gets the READY state.

Based on your input and some statistics I've found, we're good from my perspective.
Maybe we should consider mobile browsers. :thinking: 

1. https://www.w3counter.com/globalstats.php
2. https://en.wikipedia.org/wiki/Usage_share_of_web_browsers#StatCounter_(July_2008_to_August_2018)
3. https://www.stetic.com/market-share/browser/

**Comment by svettwer** [comment]
@SajanaKrishnan Safari is currently supported. If you don't specify the Endpoints remote server url, Safari will work with OSX El Capitan or greater.

**Comment by SajanaKrishnan** [comment]
Sven,

If I dont specify remote server url, how does it know where to execute

Regards
Sajana

**Comment by svettwer** [comment]
Hi!

In this case the execution would be performed locally. From your answer I infer that you want to use Safari with a remote endpoint.
This is, as you mentioned, not supported at the moment. I'm currently having a look on #497 where you addressed this issue.

BR,
Sven

**Comment by SajanaKrishnan** [comment]
Ok thanks I wanted safari executed in both local and remotewebdriver

Thanks
